### PR TITLE
feat(decide): clone user-context before calling optimizely decide

### PR DESCRIPTION
**Pull request by mnoman09**
Clone the current user-context and pass it to optimizely.decide(), decide all and in decideForKeys
- user context can be changed while processing the decide request.
- with this fix, decide can return the copy of the original user-context as a part of the decision

### CODE REVIEW

File: OptimizelySDK/OptimizelyUserContext.cs [hunk 1]
```diff
@@ -46,6 +46,8 @@ public OptimizelyUserContext(Optimizely optimizely, string userId, UserAttribute
             UserId = userId;
         }
 
+        private OptimizelyUserContext Clone() => new OptimizelyUserContext(Optimizely, UserId, Attributes, ErrorHandler, Logger);
```
Comment (jaeopt) [inline OptimizelySDK/OptimizelyUserContext.cs:49]: change "Attributes" to "getAttributes()", which should be mutex protected while copying.

**Review by jaeopt — CHANGES_REQUESTED** [review]
Looks good. A change suggested.

File: OptimizelySDK/OptimizelyUserContext.cs (hunk 1 above)
Comment (msohailhussain) [reply OptimizelySDK/OptimizelyUserContext.cs:49]: Also rename it to Copy.

**Review by msohailhussain — APPROVED** [review]
lgtm. please address jae's comments.

**Review by jaeopt — APPROVED** [review]
LGTM

**Pull request merged**